### PR TITLE
Register Azure SQLServer state store

### PR DESCRIPTION
**Pull request by fbeltrao**
# Description

Register Azure SQLServer state store

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [x] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation
* [ ] Specification has been updated
* [ ] Provided sample for the feature

### CODE REVIEW

File: pkg/components/state/loader.go [hunk 1]
```diff
@@ -47,4 +48,7 @@ func Load() {
 	RegisterStateStore("gcp.firestore", func() state.Store {
 		return firestore.NewFirestoreStateStore()
 	})
+	RegisterStateStore("azure.sqlserver", func() state.Store {
+		return sqlserver.NewSQLServerStateStore()
```
Comment (yaron2) [inline pkg/components/state/loader.go]: As this is not Azure specific, can we change to just `"sqlserver"`?

Comment (fbeltrao) [reply pkg/components/state/loader.go]: Yes, it makes sense. What about the components-contrib project? Should I:
- keep as /state/azure/
- move it to /state/
- move it to /state/microsoft/
- move it to /state/sql/sqlserver

Comment (yaron2) [reply pkg/components/state/loader.go]: /state/sqlserver

**Pull request merged**